src/ms3/bs4_parser.py

```diff
@@ -2919,28 +3013,53 @@ def set_instrument(self, staff_id: Union[str, int], trackname):
             ]
         )  # which staves share this part
         if len(staves_within_part) > 0:
-            different_values_set = np.where(
-                [
-                    new_values["id"] != self.fields[staff_key]["id"]
-                    for staff_key in staves_within_part
-                ]
-            )[
-                0
-            ]  # staves of the same part with different instruments
-            if len(different_values_set) > 0:
-                damaged_staves = staves_within_part[different_values_set]
-                damaged_dict = {
-                    elem: self.fields[elem]["id"] for elem in damaged_staves
-                }
+            damaged_upd_staves = list(
+                filter(
+                    None,
+                    [
+                        staff_key
+                        if new_values["id"] != self.updated[staff_key]
+                        else None
+                        for staff_key in set(staves_within_part) & self.updated.keys()
+                    ],
+                )
+            )
```

Comment:
Isn't that just a complicated way of saying

```python
[staff_key 
 for staff_key in set(staves_within_part) & self.updated.keys() 
 if staff_key and new_values["id"] != self.updated[staff_key]
]
 ```
 ?

Reply:
Yes, thank you, I updated in the commit [ef78db3] (https://github.com/johentsch/ms3/pull/84/commits/ef78db341032134a645c27b05301540f2c10fc8e)